### PR TITLE
Hide twitter related item if disabled

### PR DESCRIPTION
**Pull request by dadarya0**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | Yes 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
Hide following twitter related item if twitter is disabled :-

1. Social monitoring in Channel menu
2. Social tab in contact detail page
3. Social setting in global configuration
4. Social login field in form fields

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. verify following should be hidden if twitter is disabled :-

> 1. Social monitoring in Channel menu
> 2. Social tab in contact detail page
> 3. Social setting in global configuration
> 4. Social login field in form fields

### CODE REVIEW

File: plugins/MauticSocialBundle/Config/config.php [hunk 1]
```diff
@@ -46,6 +46,20 @@
...
+                    'mautic.social.integration.config',
+                ],
+            ],
+            'mautic.social.configbundle.subscriber' => [
+                'class'     => \MauticPlugin\MauticSocialBundle\EventListener\ConfigSubscriber::class,
+                'arguments' => [
+                    'mautic.social.integration.config',
+                ],
+            ],
+        ],
```
Comment (escopecz) [inline plugins/MauticSocialBundle/Config/config.php]: Let's have autowiring handle this.
```suggestion
```

File: plugins/MauticSocialBundle/Config/config.php [hunk 2]
```diff
@@ -159,6 +173,12 @@
                     'mautic.helper.integration',
                 ],
             ],
+            'mautic.social.integration.config' => [
+                'class'     => \MauticPlugin\MauticSocialBundle\Integration\Config::class,
+                'arguments' => [
+                    'mautic.helper.integration',
+                ],
+            ],
```
Comment (escopecz) [inline plugins/MauticSocialBundle/Config/config.php]:
```suggestion
```

File: plugins/MauticSocialBundle/EventListener/ConfigSubscriber.php [hunk 3]
```diff
@@ -6,10 +6,19 @@
...
 
 class ConfigSubscriber implements EventSubscriberInterface
 {
+    private Config $config;
+
+    public function __construct(
+        Config $config
+    ) {
+        $this->config = $config;
+    }
```
Comment (escopecz) [inline plugins/MauticSocialBundle/EventListener/ConfigSubscriber.php]: PHP8 🪨 
```suggestion
    public function __construct(private Config $config)
    {
    }
```

File: plugins/MauticSocialBundle/EventListener/FormSubscriber.php [hunk 4]
```diff
@@ -5,10 +5,19 @@
...
 
 class FormSubscriber implements EventSubscriberInterface
 {
+    private Config $config;
+
+    public function __construct(
+        Config $config
+    ) {
+        $this->config = $config;
+    }
```
Comment (escopecz) [inline plugins/MauticSocialBundle/EventListener/FormSubscriber.php]:
```suggestion
    public function __construct(private Config $config)
    {
    }
```

File: plugins/MauticSocialBundle/Integration/Config.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+use Mautic\PluginBundle\Helper\IntegrationHelper;
+
+final class Config
+{
+    private IntegrationHelper $integrationsHelper;
+
+    public function __construct(IntegrationHelper $integrationsHelper)
+    {
+        $this->integrationsHelper = $integrationsHelper;
+    }
```
Comment (escopecz) [inline plugins/MauticSocialBundle/Integration/Config.php]:
```suggestion
    public function __construct(private IntegrationHelper $integrationsHelper)
    {
    }
```

**Review by escopecz — CHANGES_REQUESTED** [review]
Looks good! We can delete a bunch of code due to autowiring and PHP8 syntax

**Review by escopecz — DISMISSED** [review]
Thank you Saurabh! The code looks great!

**Review by RCheesley — CHANGES_REQUESTED** [review]
Thanks for this PR, I'm always keen for us to remove stuff if it's not needed and I agree it's great that if we don't have the twitter plugin enabled that it should not be showing in the interface.  Thanks also to @escopecz for the thorough code review!

The only thing that was a bit concerning or confusing for me was when I went to a contact who had no social media profiles present,  the social tab was there but there was no content which just looked a bit untidy:

screenshot-8080-dadarya0-mautic-22m56ofd6ra ws-eu105 gitpod io-2023 10 27-14_10_44

Should we not remove the whole tab if there are no social profiles entered on the contact profile, instead of having the tab there but having it completely empty when clicked on?

**Comment by kuzmany** [comment]
@RCheesley In my opinion, your request should cover separate PR. I recommend moving this to Ready to Commit instead of blocking it as not non-related request.

**Comment by RCheesley** [comment]
@kuzmany I disagree. Per your acceptance criteria:

> 2\. verify following should be hidden if twitter is disabled :-
> 
> > 2. Social tab in contact detail page

The social tab is **not** being hidden on the contact detail page, where it should be, according to the test instructions, if the details are not present on the contact profile.

**Comment by dadarya0** [comment]
> @kuzmany I disagree. Per your acceptance criteria:
> 
> > 2. verify following should be hidden if twitter is disabled :-
> > > 2. Social tab in contact detail page
> 
> The social tab is **not** being hidden on the contact detail page, where it should be, according to the test instructions, if the details are not present on the contact profile.

@RCheesley in this PR description, I was not referring  `Social tab` in field , I was referring `Social tab` in lower part of detail page where we have History, audit log, Integration etc because it was related to twitter configuration.

**Review by RCheesley — APPROVED** [review]
Ahhhh, OK that makes sense! Screenshots ftw, could you include them in future PRs to make it easier for testers to understand what you're referring to, perhaps? Thanks!

**Review by escopecz — APPROVED** [review]

**Pull request merged**